docs/v3/advanced/api-client.mdx

```diff
@@ -43,6 +43,46 @@ with get_client(sync_client=True) as client:
 
 </CodeGroup>
 
+## Configure custom headers
+
+You can configure custom HTTP headers to be sent with every API request by setting the `PREFECT_CLIENT_CUSTOM_HEADERS` setting. This is useful for adding authentication headers, API keys, or other custom headers required by proxies, CDNs, or security systems.
+
+### Setting custom headers
+
+Custom headers can be configured via environment variables or settings. The headers are specified as key-value pairs in JSON format.
+
+<CodeGroup>
+
+```bash Environment variable
+export PREFECT_CLIENT_CUSTOM_HEADERS='{"CF-Access-Client-Id": "your-client-id", "CF-Access-Client-Secret": "your-secret"}'
+```
+
+```bash CLI
+prefect config set PREFECT_CLIENT_CUSTOM_HEADERS='{"CF-Access-Client-Id": "your-client-id", "CF-Access-Client-Secret": "your-secret"}'
+```
+
+```toml prefect.toml
+[client]
+custom_headers = '''{
+    "CF-Access-Client-Id": "your-client-id",
+    "CF-Access-Client-Secret": "your-secret",
+    "X-API-Key": "your-api-key"
+}'''
```

Comment:
Configuring custom headers via a JSON string in TOML feels a little janky. Is it possible to configure custom headers via vanilla TOML?